### ISSUE TITLE
Unexpected None values generated with xml xpaths

### ISSUE THREAD

**Issue by fcharras** [morph-kgc/morph-kgc#297]
####  What Happens?
At the time of executing [This instruction](https://github.com/morph-kgc/morph-kgc/blob/main/src/morph_kgc/materializer.py#L37) the dataframe `data` can contain `None` values, typically it happens when using xml and xpath. This step will transform the None objects into "None" strings, resulting later on in triples built from those `None` strings, while one would expect missing data to not cause any creation of triples. Adding `None` to the `na_values` config is not a satisfying workaround since users might not want to have all their actual "None" strings in the data always cast to na_values. A proper fix might be adding `data = data.dropna(axis=0, how="any", subset=list(references))` before the `map`. Or, if the `map` is not necessary (as the `# TODO: can this be removed?` suggests), removing it would also likely fix the bug.

#### To Reproduce
Basically parsing xml with some xpath that sometimes have text value, sometimes not, will generate None value. I unfortunately don't have time to craft a proper reproducer at the moment, I hope the description I provide can help highlighting the issue. Will come back at it once possible.

#### Environment (please complete the following information):
 - OS: linux
 - Python version: 3.12
 - Morph-KGC version: 2.8.0

**Comment by arenas-guerrero-julian** [comment]
I think the first solution you suggest solves the issue, could you contribute a PR?
